andela/team-odd-bn-backend
# #169258639 notifications of new booking and travel request

#### What does this PR do?
- This PR adds a functionality that will enable the travel admin to receive notification when a booking is made or the manager to receive the notification when a new trip request is made.
- This PR also adds a functionality in which users can mark notifications as read
#### Description of Task to be completed?
- Created two two routes, one for marking the all notifications as read and the other one for marking a single notification as read
- use socket IO to send notifications to the clients
- Add a user ID column in the accommodation table and bookingId in the notification table
#### How should this be manually tested?
- Open your terminal
- `git clone https://github.com/andela/team-odd-bn-backend.git`
- `cd team-odd-bn-backend`
- `git checkout ft-notification-booking-travel-request-169258639`
- Make sure your `.env` file is updated.
- In the browser, open `localhost:3000/static` (make sure you frequently reload it as the tests are running)
- Run `npm test` to run the tests and view the browser's console.
- Open postman and test the endpoints`api/v1/notifications/markRead` and `api/v1/notifications/markRead/:notificationId`
- You can also test if notifications are sent to the client from postman by creating a new trip or making a booking
#### Any background context you want to provide?
**The input data**
`
{
   "notificationIds":
   [1,2,3,4,5,11]
}
`
#### What are the relevant pivotal tracker stories?
[#169258639](https://www.pivotaltracker.com/n/projects/2407428/stories/169258639)
#### Screenshots (if appropriate)
**Data sent to the client**
<img width="1440" alt="Screen Shot 2019-12-11 at 17 08 57" src="https://user-images.githubusercontent.com/45129725/70633280-f6a55c80-1c38-11ea-927c-2bbf0d419198.png">

**Successful mark notifications as read**
<img width="687" alt="Screen Shot 2019-12-11 at 17 17 22" src="https://user-images.githubusercontent.com/45129725/70634874-a4196f80-1c3b-11ea-89a7-7fa74967ae64.png">

**String notification ID provided in the array**
<img width="456" alt="Screen Shot 2019-12-11 at 17 18 16" src="https://user-images.githubusercontent.com/45129725/70634875-a4b20600-1c3b-11ea-8787-dd103246a6cf.png">


**String notification ID provided in the params**
<img width="482" alt="Screen Shot 2019-12-11 at 17 22 39" src="https://user-images.githubusercontent.com/45129725/70634879-a54a9c80-1c3b-11ea-928d-21081b5ffb49.png">

#### Questions:
